ext/src/ruby_api/config.rs

```diff
@@ -25,13 +25,17 @@ define_rb_intern!(
     WASM_MEMORY64 => "wasm_memory64",
     PROFILER => "profiler",
     CRANELIFT_OPT_LEVEL => "cranelift_opt_level",
+    STRATEGY => "strategy",
```

Comment:
I've opted to not put the parts outside the conditional behind a `cfg` flag because it makes integrating support into the conditional much more straightforward. But if you'd like, I can change the conditional to use an early return or look into refactoring it into a match statement (or another option) instead so it's easier to put more of the plumbing for the strategy option behind the `cfg` flag.